tests/utils.py

```diff
@@ -60,36 +61,40 @@ class RemoteOpenAIServer:
 
     def __init__(self,
                  model: str,
-                 cli_args: List[str],
+                 serve_args: List[str],
                  *,
                  env_dict: Optional[Dict[str, str]] = None,
                  auto_port: bool = True,
                  max_wait_seconds: Optional[float] = None) -> None:
-        if not model.startswith("/"):
-            # download the model if it's not a local path
-            # to exclude the model download time from the server start time
-            snapshot_download(model)
         if auto_port:
-            if "-p" in cli_args or "--port" in cli_args:
+            if "-p" in serve_args or "--port" in serve_args:
                 raise ValueError("You have manually specified the port"
                                  "when `auto_port=True`.")
 
-            cli_args = cli_args + ["--port", str(get_open_port())]
+            serve_args = serve_args + ["--port", str(get_open_port())]
 
         parser = FlexibleArgumentParser(
             description="vLLM's remote OpenAI server.")
         parser = make_arg_parser(parser)
-        args = parser.parse_args(cli_args)
+        args = parser.parse_args(["--model", model, *serve_args])
         self.host = str(args.host or 'localhost')
         self.port = int(args.port)
 
+        # download the model before starting the server to avoid timeout
+        engine_args = AsyncEngineArgs.from_cli_args(args)
+        engine_config = engine_args.create_engine_config()
```

Comment:
can we directly create a load config object? it should be simple in my opinion. just load format auto.

Reply:
Let's fix the tests first before simplifying this.

Reply:
I'm not that familiar with the model loading code so may need some help here.